Rebranding of Avaya BOSS and VOSS to Extreme

ospfbgp opened the pull request:
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/50c6d700-1953-4326-b026-a7ff8ed18300) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/50c6d700-1953-4326-b026-a7ff8ed18300) completed: **No new issues**

ospfbgp commented:
Since Extreme bought the networking assets from Avaya they need to have new logos/images for products.  The products are VOSS and BOSS operating systems.  I have created new logos for easy viewing on devices pages.  I put VOSS and BOSS below Extreme images.  I also removed an OID for a device from VOSS(avaya-vsp.yaml).  The device is .1.3.6.1.4.1.45.3.79.1 (VSP 7024XLS).  Even though it is call VSP it is a BOSS/(Ethernet routing switch).  I have fixed this and done testing.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9251eac6-749b-434a-85fb-bbd73743792d) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9251eac6-749b-434a-85fb-bbd73743792d) completed: **No new issues**

ospfbgp commented:
laf.  I went to the link, uploaded the files and now the svg files are 523 bytes and 520 bytes.  I hope this is ok now.

BOSS/VOSS os history
Extreme Ethernet Routing Switch(ERS) use the BOSS(Baystack operating system software).  This is all the old Synoptics/Nortel/Avaya switches called Ethernet Routing Switch. There was a single VSP(Virtual Switch Platform) that used the BOSS and it was the VSP-7024.  I have no idea why they did this other than marketing.  It made no sense IMO.  All Avaya/Extreme VSP use the VOSS(VSP operating system software).

Do you want me to change the following files for avaya-ers?  If so do you want me to make them extreme-boss or extreme-ers?
./discovery/vlans/avaya-ers.inc.php
./discovery/mempools/avaya-ers.inc.php
./discovery/sensors/temperature/avaya-ers.inc.php
./discovery/sensors/state/avaya-ers.inc.php
./polling/os/avaya-ers.inc.php
./polling/mempools/avaya-ers.inc.php
./definitions/avaya-ers.yaml

Do you want me to change the following files for avaya-vsp? If so do you want me to make them extreme-vsp or extreme-voss?
./definitions/discovery/avaya-vsp.yaml
./definitions/avaya-vsp.yaml

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/30c08e3f-2719-49a3-b472-d16c1c903e43) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/3f20fc57-9b9c-4de0-89fc-68be03cf0d3a) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ca3a12e9-e53a-4933-a746-61dd33488b7d) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/da39e7c0-fbc7-4b9b-98c0-ef2c13392f22) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/27f4ebb3-ea4f-4b5b-a79e-e4e16d2bcfaa) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e0f9a226-591b-43fc-af68-5c60fc564407) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/58144eed-bbaa-4552-b845-7312be5bab6d) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/11a5a47f-b81f-4087-ae69-e0a12e9b8cee) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/d29f2023-192d-4e1c-8dd0-3db6581408d0) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/05f692e5-7ab8-4fff-bcc0-93da73100a30) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/da39e7c0-fbc7-4b9b-98c0-ef2c13392f22) completed: **No new issues**

ospfbgp commented:
I have renamed the files from avaya-ers -> extreme-boss and avaya-vsp -> extreme-voss.  I have also updated all the files to reflect the new reference.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/05f692e5-7ab8-4fff-bcc0-93da73100a30) completed: **No new issues**

murrant commented:
How about renaming them to ers, voss, and boss?

ospfbgp commented:
So here is the breakdown.

BOSS = ERS3XXX, ERS4xxx, ERS5xxx,  VSP7024XX
This includes the Nortel and Avaya ERS platform.  The OS for these platforms has always been called BOSS and uses the same mibs for fans, cpu, memory, temperature.

VOSS = VSP44xx, VSP48XX, VSP72XX, VSP82XX, VSP84XX, VSP86XX, VSP88XX
All VOSS use the same mibs for fans, cpu, memory, temperature

Extreme bought all the networkings assets and that includes the ERS and VOSS from Avaya.

I feel it would be very difficult to categorize these into ERS, VOSS, and BOSS.   ERS and BOSS are going to share the same mibs.  ERS and BOSS would end up being identical.  I can get a list of OIDs that fit BOSS exactly if needed but I do not see why when we can just use ERS.  Any recent products over the last 5-6 years fall into the BOSS and VOSS.  Avaya/Nortel end of life and end of support all the ES platform, ERS86xx, and ERS88xx.  You can not even get parts for these platforms without going through a third parth.

murrant commented:
So, you want just boss and voss?  Sounds ok to me.  I was just saying there is no reason to include the company name in the os.  We do that in the description.

ospfbgp commented:
All fixed with VOSS and BOSS

murrant commented:
Sorry, I meant the reverse of what you have done.  files named boss/voss, text as "Extreme BOSS"/"Extreme VOSS"

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/364b4b3b-4642-4d00-8fd8-fe7b7cfffd65) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ce2939d9-1c3d-41fb-ab48-ce2699eccf6b) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b21069dd-5d26-44f3-9049-6425ebcc8e03) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4f712bcf-dfa1-4d3e-9f83-85305fe52e97) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/118459a5-378f-455d-8915-96edca4a74b1) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/c858faab-b9b3-4aef-8df2-0dc018740a73) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/027af107-ef95-4ade-9188-ff7f80a41582) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ce2939d9-1c3d-41fb-ab48-ce2699eccf6b) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4b97becd-0258-436e-abba-d53e55fe06ae) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/1e8f85cd-ad00-410c-a467-a846e1302475) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/67af719c-4cc9-4259-96a2-48007f119cfb) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/a93d8298-39d3-4046-9baa-1a2a327e7905) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4dfb21b3-47c9-4570-928f-678fc88447b4) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/d86225c8-79b1-4713-834f-da6398863832) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ad4018a7-4a67-4a2f-bf24-85d89b5a371d) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/67af719c-4cc9-4259-96a2-48007f119cfb) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/00824316-5b88-4eb5-9b8f-bfa5312aaf5c) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/221d6e55-844a-49dd-b983-4ab6e4e84798) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e2652604-3d0b-49b0-ac37-3007abf3e970) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/00824316-5b88-4eb5-9b8f-bfa5312aaf5c) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/f2c5b91e-8744-4d0e-bffc-2aaca53c8678) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ba1288e2-382e-4398-8dea-77c739312006) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9ed2aff8-1c3b-4410-89f5-3dd7c8c7b143) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e2652604-3d0b-49b0-ac37-3007abf3e970) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/cb86e78b-bdf7-45a4-8386-331a0f43bbc6) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/264e60fe-8d75-4694-80be-480ac59234ed) was created.

ospfbgp commented:
I have cleaned up all the files now.  I think everything is correct.  Renamed os to voss and boss.  I also renamed all the files to voss and boss.  I changed all references from avaya-ers to boss.  I have changed all references from avaya-ers to voss.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9ed2aff8-1c3b-4410-89f5-3dd7c8c7b143) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ae2a9c76-45ea-434f-bfa9-aa290af65208) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/264e60fe-8d75-4694-80be-480ac59234ed) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/ae2a9c76-45ea-434f-bfa9-aa290af65208) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/7b54ec9f-c9cd-4f92-8295-6d38ac237571) completed: **No new issues**

laf commented on includes/discovery/mempools/boss.inc.php:
```diff
@@ -2,14 +2,14 @@
         $t   = explode(' ', $t);
         $oid = str_replace($OID, '', $t[0]);
-        discover_mempool($valid_mempool, $device, $oid, 'avaya-ers', 'Unit '.($i + 1).' memory', '1', null, null);
+        discover_mempool($valid_mempool, $device, $oid, 'boss', 'Unit '.($i + 1).' memory', '1', null, null);
```
I think these will need to stay the same to avoid losing data, same for all the `discover_` functions.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/f523ae40-9d71-4e52-928c-961257e5031f) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/f0888dc5-af83-49c1-b23f-dc87641293e6) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/2fca5245-bfe9-4937-ac58-28e9fe4d263a) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/fc1d89d9-fe0c-46c3-9b24-261feb41f15c) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b8579db1-2854-4c87-b6cc-533b39b82ed2) was created.

ospfbgp commented:
I have changed discovery functions to still use avaya-vsp and avaya-ers.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/f0888dc5-af83-49c1-b23f-dc87641293e6) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b8579db1-2854-4c87-b6cc-533b39b82ed2) completed: **No new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/2a2b1319-49be-4fa5-bf4b-98569fb94f8d) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/bbc63909-48ca-4e2d-b9fa-2d6d0e03c508) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/37c3ce92-68cd-4930-b053-68c70490c907) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/acaa5dc7-d48c-47e4-a26e-b4c8ec61e394) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4316d80c-b502-44fd-b03e-9be688747201) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/1e98fefb-97cb-4f22-9ca5-c975e3fffd6f) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/006edb3e-471f-4093-8ac1-a20cd4d241a9) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/81530f28-a809-4ac8-bcd3-eecade1b4c2d) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/7da1cde5-dd7c-49ad-aa75-059aaf710fb3) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9e211742-7f37-4114-8fa4-0764ae051991) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/bae7dbf8-9b32-497e-bfb0-b84e22bb3dfe) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/21cce119-b6c6-4b10-89fe-916a4d2ee723) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4316d80c-b502-44fd-b03e-9be688747201) completed: **No new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/21cce119-b6c6-4b10-89fe-916a4d2ee723) completed: **No new issues**

ospfbgp commented:
I have updated tests/snmpsim and tests/data.

laf commented:
@ospfbgp I can't merge this at present as the unit tests are failing. Can you rebase against master.

ospfbgp commented:
Not sure how to do that. I am using the web GitHub. 

Sent from my iPhone

> On Apr 15, 2018, at 11:25 AM, Neil Lathwood <notifications@github.com> wrote:
> 
> @ospfbgp I can't merge this at present as the unit tests are failing. Can you rebase against master.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/443742a6-9174-4a0e-ad3a-f9e46c20284e) was created.

ospfbgp commented:
I have rebased I believe.  I am close.  I should be up to date now.   Let me know if note but I am very close.

Cox-MacBook-Pro:librenms danielcox$ git remote -v
origin	https://github.com/ospfbgp/librenms (fetch)
origin	https://github.com/ospfbgp/librenms (push)
upstream	https://github.com/librenms/librenms.git (fetch)
upstream	https://github.com/librenms/librenms.git (push)
Cox-MacBook-Pro:librenms danielcox$ git rebase master
Current branch master is up to date.

laf commented:
Ok I think it's failing due to `LibreNMS/OS/AvayaErs.php`. Need to rename that file + the function within it.

`$this->getName(),` will probably need updating to `avaya-ers` so old data isn't lost

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/00a052ea-0da7-4437-92ef-c1a96cdda4f3) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/00a052ea-0da7-4437-92ef-c1a96cdda4f3) completed: **No new issues**

ospfbgp commented:
updated let me know if it works.  I did not know the processor discovery was there. I was wondering where it was for the boss because it worked.  I know when I did processor for voss I put processor discover in /librenms/includes/definitions/discovery

laf commented:
@ospfbgp You need to rename the class as well https://github.com/ospfbgp/librenms/blob/9e6f476c875501f4d58f3ee675276aaa645c024c/LibreNMS/OS/Boss.php#L32

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/fc2a3d70-41a8-4229-9ec1-910b0125302d) was created.

ospfbgp commented:
A lot of little changes.  Always the little things.
Thanks for you patience..    :-)

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/fc2a3d70-41a8-4229-9ec1-910b0125302d) completed: **1 new issues, 2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e33b7b0f-cd3f-4300-9f65-58051fd44f4a) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e33b7b0f-cd3f-4300-9f65-58051fd44f4a) completed: **1 new issues, 2 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/dbbbbccf-dcc5-4494-b5a3-ff5b3968a3eb) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/dbbbbccf-dcc5-4494-b5a3-ff5b3968a3eb) completed: **1 new issues, 2 updated code elements**

murrant commented on html/images/os/extremeboss.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" version="1" width="100" height="100"><g clip-rule="evenodd" fill="#6502a2" fill-rule="evenodd" stroke-width="1.46"><path d="M95 16.04V0H4.918v63.96l19.303-16.04V16.04z"/><path d="M82.131 47.92V31.882H43.286v16.04zM24.221 80H95V63.96H4.918z"/></g><text style="line-height:1.25;text-orientation:upright" x="20.998" y="102.9" transform="scale(1.03945 .96205)" fo ...
```
You need to drop length/width and add viewbox so SVGs render properly in IE.  Thanks

murrant commented on html/images/os/extremevoss.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" version="1" width="100" height="100"><g clip-rule="evenodd" fill="#6502a2" fill-rule="evenodd" stroke-width="1.46"><path d="M95 16.04V0H4.918v63.96l19.303-16.04V16.04z"/><path d="M82.131 47.92V31.882H43.286v16.04zM24.221 80H95V63.96H4.918z"/></g><text style="line-height:1.25;text-orientation:upright" x="23.541" y="101.295" transform="scale(1.02324 .9773)" f ...
```
You need to drop length/width and add viewbox so SVGs render properly in IE.  Thanks

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b272ab12-340a-4537-836d-3da593a054cd) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e3cfc26a-9fef-4113-9a31-f1431ffc71b8) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b272ab12-340a-4537-836d-3da593a054cd) completed: **1 new issues, 2 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e3cfc26a-9fef-4113-9a31-f1431ffc71b8) completed: **1 new issues, 2 updated code elements**

ospfbgp commented:
Thank you for all the help.
